src/Microsoft.AspNet.Hosting.Server.Abstractions/IHttpApplication.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Features;
+
+namespace Microsoft.AspNet.Hosting.Server
+{
+    /// <summary>
+    /// Represents an HttpApplication.
+    /// </summary>
+    public interface IHttpApplication
```

Comment:
I think HttpRequestHandler is better, since its not really clear what exactly an HttpApplication is doing :)  Instead of, well duh, this handles the request!